frameworks/kafka/docs/security.md

```diff
@@ -0,0 +1,242 @@
+example/kafka-2-broker.agoodexample.autoip.dcos.thisdcos.directory@EXAMPLE
+```
+
+#### Place Service Keytab in DC/OS Secret Store
+
+The DC/OS Apache Kafka service uses a keytab containing all broker principals (service keytab) to simplify orchestration. After creating the principals above, generate the service keytab making sure to include all the broker principals. This will be stored as a secret in the DC/OS Secret Store.
+
+*Note*: DC/OS 1.10 does not support directly adding binary secrets to the secret store, only text files are supported. Instead, first base64 encode the file, and save it to the secret store as `/desired/path/__base_64__secret_name`. The DC/OS security modules will handle decoding the file when it is used by the service. More details [here](https://docs.mesosphere.com/services/ops-guide/overview/#binary-secrets).
+
+The service keytab should be stored at `service/path/service.keytab` (as noted above for 1.10, it would be `__base_64__service.keytab`), where `service/path` matches the path of the service. For example, if installing with the options
+```
+{
+    "service": {
+        "name": "a/good/example"
+    }
+}
+```
+then the service keytab should be stored at `a/good/service.keytab`.
+
+Documentation for adding a file to the secret store can be found [here](https://docs.mesosphere.com/latest/security/ent/secrets/create-secrets/#creating-secrets-from-a-file-via-the-dcos-enterprise-cli).
+
+*Note*: Secrets access is controlled by [DC/OS Spaces](https://docs.mesosphere.com/1.10/security/ent/#spaces-for-secrets). Technically, any secret path "above" that of the service will be accessible by the service. Matching the two paths is, however, the most secure option. Additionally `service.keytab` is a convention and not a requirement.
+
+
+#### Install the service
+
+Install the DC/OS Apache Kafka service will the following options alongside your own:
+```
+{
+    "service": {
+        "kerberos": {
+            "enabled": true,
+            "enabled_for_zookeeper": <true|false default false>
+            "kdc": {
+                "hostname": "<kdc host>",
+                "port": <kdc port>
+            },
+            "primary": "<default kafka>",
+            "realm": "<realm>",
+            "keytab_secret": "<path to keytab>",
+            "debug": <true|false default false>
+        }
+    }
+}
+```
+
+*Note*: If `service.kerberos.enabled_for_zookeeper` is set to true, then the additional setting
+```
+{
+    "kafka": {
+        "kafka_zookeeper_uri": <list of zookeeper hosts>
+    }
+}
+```
+`kafka.kafka_zookeeper_uri` must be configured to point at a Kerberized Apache ZooKeeper. The DC/OS Apache Zookeeper service is intended for this purpose and supports Kerberos.
+
+*Note*: It is possible to enable Kerberos after initial installation but the service may be unavailable during the transition. Additionally, your Kafka clients will need to be reconfigured.
+
```

Comment:
There is no section on authenticating a Kerberos client.

Reply:
My vote: Tough, that's on you.

I only mention anything for SSL because you need to do something interacting _with_ the DC/OS ca.

I guess we could link to docs on authenticating a kerberized client from some other source.

Reply:
It works, because we send the broker count to all the brokers (so they can calculate their super users) which leads to a rolling restart before the scale up is done.

Will need to make sure we _dont_ do this if SSL Auth is not enabled.